pkg/executor/util.go

```diff
@@ -24,6 +24,11 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
+// We limit the statefulset, pod, service and pvc name to be lesser than 46
+// characters to accomodate for the controller revision hash and the unique
+// identifier that kubernetes adds while creating the statefulset pods.
```

Comment:
I had some comments/questions earlier about how we should handle the fact that this number depends on the number of digits in maxKDMembers ... what were your thoughts on that?

Reply:
Also I still think it would be good to have more comments here about how this is calculated. I.e. I've already forgotten enough about the controller revision hash thing to understand why this is 46.

To recap/re-figure-this-out:

I know that one of our limitations is that service names have to be 63 max, and we name our services after the statefulset members which have an index # appended. The length of the GenerateName UID is 5 chars, the current max number of digits in maxKDMembers is 4, plus a hyphen between them, so our GenerateName prefix can be at most 53 to deal with that limitation.

(The statefulset.kubernetes.io/pod-name label value would also probably give us the same sort of limit.)

We also have the controller revision hash label value as a limitation. A label value can be 63 max. Similar calculation as above except instead of an index number there will be a 10 character hash appended. That would seem to indicate that our max size would be 63 - (5 + 1 + 10) = 47.

Since your loop below does "less than" and not "less than or equal to", you're currently using a limit of 45.

Obviously a smaller limit is safe too but the discrepancy does make me think that one of us is overlooking something about what the limit(s) really are. We need to be extra sure we understand what is driving the limits and then add comments here to explain.

Reply:
> I had some comments/questions earlier about how we should handle the fact that this number depends on the number of digits in maxKDMembers ... what were your thoughts on that?

Do we have an open issue that talks about changing the member size to be dynamic? If yes, I can add a comment there to address the naming scheme based on the approach that we take there. If not, I can open an issue and add a comment there. As for now, since we are already checking on the current member size, we should be fine? I'll add a comment explaining the max nameLengthLimit calculation.

Let me know if this sounds okay to you.

Reply:
> Also I still think it would be good to have more comments here about how this is calculated. I.e. I've already forgotten enough about the controller revision hash thing to understand why this is 46.
> 
> To recap/re-figure-this-out:
> 
> I know that one of our limitations is that service names have to be 63 max, and we name our services after the statefulset members which have an index # appended. The length of the GenerateName UID is 5 chars, the current max number of digits in maxKDMembers is 4, plus a hyphen between them, so our GenerateName prefix can be at most 53 to deal with that limitation.
> 
> (The statefulset.kubernetes.io/pod-name label value would also probably give us the same sort of limit.)
> 
> We also have the controller revision hash label value as a limitation. A label value can be 63 max. Similar calculation as above except instead of an index number there will be a 10 character hash appended. That would seem to indicate that our max size would be 63 - (5 + 1 + 10) = 47.
> 
> Since your loop below does "less than" and not "less than or equal to", you're currently using a limit of 45.
> 
> Obviously a smaller limit is safe too but the discrepancy does make me think that one of us is overlooking something about what the limit(s) really are. We need to be extra sure we understand what is driving the limits and then add comments here to explain.

I have tried to explain the calculation. Let me know if it makes sense to you.

Reply:
> Do we have an open issue that talks about changing the member size to be dynamic? If yes, I can add a comment there to address the naming scheme based on the approach that we take there. If not, I can open an issue and add a comment there.

We don't have such an issue; go ahead and create it.